### ISSUE TITLE
[Feature Request] More status codes

### ISSUE THREAD

**Issue by lonix1** [certbot/certbot#9321]
Certbot returns `0` and `1` status codes - from the [docs say](https://eff-certbot.readthedocs.io/en/stable/using.html#certbot-command-line-options):
> certbot renew exit status will only be 1 if a renewal attempt failed. This means certbot renew exit status will be 0 if no certificate needs to be updated. If you write a custom script and expect to run a command only after a certificate was actually renewed you will need to use the --deploy-hook since the exit status will be 0 both on successful renewal and when renewal is not necessary.

I'm using a hook, but sometimes I only want to know whether the cert was renewed, so I can take some action. Add docker and other tools into the mix, and it gets messy for something so simple.

It would be nice to have more status codes:
- `0` = "renewal unnecessary" (backwards compatible)
- `1` = "renewal failed" (backwards compatible)
- `2` = "renewal succeeded" (*at the very least, please add this!*)
- `10` = "too complicated, use a hook instead" (a catch-all when processing multiple certs)

Please at least give us the basics, which includes "renewal succeeded".

Thanks!

**Comment by alexzorin** [comment]
Hello,

>I'm using a hook, but sometimes I only want to know whether the cert was renewed, so I can take some action

Could you elaborate on why `--deploy-hook` (or `/etc/letsencrypt/renewal-hooks/deploy/`) doesn't fit here?

**Comment by lonix1** [comment]
It does fit, and I use it without problems.

But... it's a long way of doing something very simple. In a dockerised environment which is managed by a config management system (e.g. ansible), it means I need a template for a hook file, and a task to upload such files to remote servers, and to mount such hooks in containers. And those hooks just signal other processes that a cert was renewed. And those also have scripts which must be mounted, etc.

When all one wants to know is that a process succeeded, the idiomatic way is to return an appropriate status code. That would eliminate a lot of busywork (which is not just a pain, but adds complexity to maintenance).

Hope I've explained properly. Thanks for your response.
